### PR TITLE
chore: add support for python 3.12

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
Adds support for python 3.12. 
Also updates a test with a race condition in 3.12

### CODE REVIEW

**Review by hsong-rh — APPROVED** [review]

File: setup.cfg [hunk 1]
```diff
@@ -16,6 +16,7 @@ classifiers =
 	Programming Language :: Python :: 3.9
 	Programming Language :: Python :: 3.10
 	Programming Language :: Python :: 3.11
+    Programming Language :: Python :: 3.12
```
Comment (jshimkus-rh) [inline setup.cfg:19]: This line should use leading tab as the previous lines do.

Comment (Alex-Izquierdo) [reply setup.cfg:19]: I don't know where might come the mistake, but it is actually correctly indented.

Comment (Alex-Izquierdo) [reply setup.cfg]: updated.

**Comment by Alex-Izquierdo** [comment]
@hsong-rh @msmagnanijr please review again, I had to change spaces by tabs

**Review by jshimkus-rh — APPROVED** [review]

**Review by hsong-rh — APPROVED** [review]

**Pull request merged**